controllers/managedocs_controller.go

```diff
@@ -920,6 +936,41 @@ func (r *ManagedOCSReconciler) reconcileOnboardingValidationSecret() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileCSIKMSConnectionDetailsConfigMap() error {
+	if r.DeploymentType != consumerDeploymentType {
+		r.Log.Info(fmt.Sprintf("Non consumer deployment, skipping reconcile for %s config map", csiKMSConnectionDetailsConfigMapName))
+		return nil
+	}
+	r.Log.Info(fmt.Sprintf("Reconciling %s configMap", csiKMSConnectionDetailsConfigMapName))
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.csiKMSConnectionDetailsConfigMap, func() error {
+		if err := r.own(r.csiKMSConnectionDetailsConfigMap); err != nil {
+			return err
+		}
+
+		encryptionMetadata := struct {
+			EncryptionKMSType string `json:"encryptionKMSType"`
+			SecretName        string `json:"secretName"`
+		}{
+			EncryptionKMSType: "aws-sts-metadata",
+			SecretName:        "tenant-aws-secret", // user needs to create a secret with this name
```

Comment:
The feature is in Dev preview. So no docs needed

Reply:
Yes, it can be referred here
https://github.com/ceph/ceph-csi/blob/7b44054a7ec17e81530d12cffa56070d9bdc6af3/examples/kms/vault/csi-kms-connection-details.yaml#L75